src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/ProviderHandler.java

```diff
@@ -146,14 +147,15 @@ public static void removeProviderInternal(final IStandardRequestManager manager,
                         if (objectToken instanceof IToken)
                         {
                             final IToken<?> childToken = (IToken<?>) objectToken;
+                            // rawtype because of java generics
                             final IRequest childRequest = RequestHandler.getRequest(manager, childToken);
 
                             //Check if the child has been assigned. If not, no work done, no cleanup needed.
                             if (RequestHandler.isAssigned(manager, childToken))
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 5 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS134)